Actions/AL-Go-Helper.ps1

```diff
@@ -258,70 +279,102 @@ function DownloadAndImportBcContainerHelper {
+            }
+            else {
+                Write-Host "Downloading BcContainerHelper $bcContainerHelperVersion version from CDN"
+                $webclient.DownloadFile("https://bccontainerhelper.azureedge.net/public/$($bcContainerHelperVersion).zip", "$tempName.zip")
+            }
         }
         Expand-7zipArchive -Path "$tempName.zip" -DestinationPath $tempName
-        $BcContainerHelperPath = (Get-Item -Path (Join-Path $tempName "*\BcContainerHelper.ps1")).FullName
+        $bcContainerHelperPath = (Get-Item -Path (Join-Path $tempName "*\BcContainerHelper.ps1")).FullName
         Remove-Item -Path "$tempName.zip" -ErrorAction SilentlyContinue
-    }
-    . $BcContainerHelperPath @params
-    $tempName
-}
-
-function CleanupAfterBcContainerHelper {
-    Param(
-        [string] $bcContainerHelperPath
-    )
-
-    if ($bcContainerHelperPath) {
-        try {
-            Write-Host "Removing BcContainerHelper"
-            Remove-Module BcContainerHelper
-            Remove-Item $bcContainerHelperPath -Recurse -Force
+        if ($bcContainerHelperVersion -notlike "https://*") {
+            # Check whether the version is already available in the cache
+            $version = Get-Content -Encoding UTF8 -Path (Join-Path $tempName 'BcContainerHelper/Version.txt')
+            $cacheFolder = Join-Path $bcContainerHelperRootFolder $version
+
+            # To avoid two agents on the same machine downloading the same version at the same time, use a mutex
+            $buildMutexName = "DownloadAndImportBcContainerHelper"
+            $buildMutex = New-Object System.Threading.Mutex($false, $buildMutexName)
+            try {
+                try {
+                    if (!$buildMutex.WaitOne(1000)) {
+                        Write-Host "Waiting for other process loading BcContainerHelper"
+                        $buildMutex.WaitOne() | Out-Null
+                        Write-Host "Other process completed loading BcContainerHelper"
+                    }
+                }
+                catch [System.Threading.AbandonedMutexException] {
+                    Write-Host "Other process terminated abnormally"
+                }
+                if (Test-Path $cacheFolder) {
+                    Remove-Item $tempName -Recurse -Force
+                }
+                else {
+                    Rename-Item -Path $tempName -NewName $version
+                }
+            }
+            finally {
+                $buildMutex.ReleaseMutex()
+            }
+            $bcContainerHelperPath = Join-Path $cacheFolder "BcContainerHelper/BcContainerHelper.ps1"
         }
-        catch {}
     }
+    Write-Host "Import from $bcContainerHelperPath"
+    . $bcContainerHelperPath @params
+    $env:BcContainerHelperPath = $bcContainerHelperPath
+    Add-Content -Encoding UTF8 -Path $ENV:GITHUB_ENV "BcContainerHelperPath=$bcContainerHelperPath"
```

Comment:
Would we be able to avoid creating the BcContainerHelperPath environment variable if we placed bccontainerhelper in ` 'C:\ProgramData\BcContainerHelper\($Env:GITHUB_RUN_ID)\...' ` or ` 'C:\ProgramData\BcContainerHelper\($Env:GITHUB_JOB)\...' ` rather than ` 'C:\ProgramData\BcContainerHelper\<Random Guid>\...' `?